### PR TITLE
CSP-184-web-AddVenueAndDelivery

### PR DESCRIPTION
**Pull request by raviudari01**

### CODE REVIEW

File: src/SFA.DAS.Roatp.CourseManagement.Web.UnitTests/Controllers/ProviderCourseLocationAddControllerTests/ProviderCourseLocationAddPostControllerTests.cs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+            _mediatorMock.Verify(m => m.Send(It.IsAny<AddProviderCourseLocationCommand>(), It.IsAny<CancellationToken>()), Times.Once);
+
+            var actual = (RedirectToRouteResult)result;
+            Assert.NotNull(actual);
+            actual.RouteName.Should().Be(RouteNames.GetProviderCourseLocations);
+        }
+
+        [Test, AutoData]
+        public async Task Post_InValidModel_RedirectToRoute(GetAllProviderLocationsQueryResult resultAllProviderLocations,
```
Comment (DevenKShah) [inline src/SFA.DAS.Roatp.CourseManagement.Web.UnitTests/Controllers/ProviderCourseLocationAddControllerTests/ProviderCourseLocationAddPostControllerTests.cs:64]: Wrong test name

File: src/SFA.DAS.Roatp.CourseManagement.Web.UnitTests/Controllers/ProviderCourseLocationAddControllerTests/ProviderCourseLocationAddPostControllerTests.cs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+                .ReturnsAsync(resultProviderCourseLocations);
+
+            _sut.ModelState.AddModelError("key", "error");
+            var result = await _sut.SubmitAProviderlocation(larsCode, model);
+
+            _mediatorMock.Verify(m => m.Send(It.IsAny<AddProviderCourseLocationCommand>(), It.IsAny<CancellationToken>()), Times.Never);
+
+            var viewResult = result as ViewResult;
+            viewResult.Should().NotBeNull();
+            viewResult.ViewName.Should().Contain("AddTrainingCourseLocation.cshtml");
```
Comment (DevenKShah) [inline src/SFA.DAS.Roatp.CourseManagement.Web.UnitTests/Controllers/ProviderCourseLocationAddControllerTests/ProviderCourseLocationAddPostControllerTests.cs:82]: use ViewPath constant here. `Should().Be(...Controller.ViewPath)`

**Review by DevenKShah — DISMISSED** [review]

File: src/SFA.DAS.Roatp.CourseManagement.Web.UnitTests/Validators/ProviderCourseLocationAddSubmitModelValidatorTests/DeliveryMethodValidationTests.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+                result.IsValid.Should().BeTrue();
+                result.ShouldNotHaveValidationErrorFor(m => m.HasDayReleaseDeliveryOption);
+            }
+            else
+            {
+                result.IsValid.Should().BeFalse();
+                result.ShouldHaveValidationErrorFor(m => m.HasDayReleaseDeliveryOption).WithErrorMessage(ProviderCourseLocationAddSubmitModelValidator.DeliveryMethodErrorMessage);
+            }
+        }
+    }
```
Comment (MarkFCain) [inline src/SFA.DAS.Roatp.CourseManagement.Web.UnitTests/Validators/ProviderCourseLocationAddSubmitModelValidatorTests/DeliveryMethodValidationTests.cs:33]: Needs coverage for testing empty Training|VenueNavigationId

**Review by MarkFCain — DISMISSED** [review]

**Review by MarkFCain — APPROVED** [review]
